src/main/java/core/basesyntax/shop/service/impl/Report.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax.shop.service.impl;
+
+import core.basesyntax.shop.service.ParseShopData;
+import core.basesyntax.shop.service.ReadFromFile;
+import core.basesyntax.shop.service.ShopService;
+import core.basesyntax.shop.service.Validator;
+import core.basesyntax.shop.service.WriteToFile;
+import core.basesyntax.shop.strategy.FileReadStrategy;
+import core.basesyntax.shop.strategy.FileWriteStrategy;
+
+public class Report {
+    private ShopService shopService;
+    private ParseShopData parseShopData;
+    private ReadFromFile readFromFile;
+    private WriteToFile writeToFile;
+    private Validator validator;
+
+    public Report() {
+        shopService = new ShopServiceImpl();
+        validator = new ValidatorImpl();
+        parseShopData = new ParseShopDataImpl();
+    }
+
+    public boolean createReport(String fromFilename, String toFilename) {
+        readFromFile = FileReadStrategy.chooseReadFileFormat(fromFilename);
+        writeToFile = FileWriteStrategy.chooseWriteFileFormat(toFilename);
+        String table = readFromFile.read(fromFilename).toLowerCase();
```

Comment:
You should not make operations with data within this class.
Also you needn't to change data before validation.
```suggestion
        String table = readFromFile.read(fromFilename);
```